dkeyserver/Enclave/enclave.cpp

```diff
@@ -387,7 +335,368 @@ int handle_communication_until_done(
     return ret;
 }
 
-int sgx_set_up_tls_server(char *server_port)
+unsigned long inet_addr(const char *str)
+{
+    unsigned long lHost = 0;
+    char *pLong = (char *)&lHost;
+    char *p = (char *)str;
+    while (p)
+    {
+        *pLong++ = atoi(p);
+        p = strchr(p, '.');
+        if (p)
+            ++p;
+    }
+    return lHost;
+}
+
+int connect(int sockfd, const struct sockaddr *servaddr, socklen_t addrlen)
+{
+    int ret = -1;
+
+    if (ocall_connect(&ret, sockfd, servaddr, addrlen) == SGX_SUCCESS)
+        return ret;
+
+    return -1;
+}
+
+int create_socket(const char *server_name, uint16_t server_port)
+{
```

Comment:
seems that inet_addr(), connect(), create_socket() and some other functions for the socket server are duplicated, can we move them into a common file in the folder https://github.com/intel/ehsm/tree/main/utils/sgx_socket/tls

Reply:
this also need to update the file in dkeycache and core projects

Reply:
better to optimize in a new patch since linux-sgx also duplicates the codes.

Reply:
we can merge current patch in advance.